tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -27,138 +22,147 @@ describe(__filename, () => {
         {children || 'the default text of this button'}
       </ConfirmButton>,
-      ConfirmButtonBase,
     );
   };
 
-  const renderWithDialog = ({
-    store = dispatchClientMetadata().store,
-    ...otherProps
-  } = {}) => {
-    const root = render({ store, ...otherProps });
+  const renderWithDialog = ({ ...otherProps } = {}) => {
+    render(otherProps);
 
     // Click to open ConfirmationDialog.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find(ConfirmationDialog)).toHaveLength(1);
-
-    return root;
-  };
-
-  const getDialogProp = (root, propName) => {
-    const confirmDialog = root.find(ConfirmationDialog);
-    expect(confirmDialog).toHaveProp(propName);
-    return confirmDialog.prop(propName);
+    const button = screen.getByRole('button', {
+      name: 'the default text of this button',
+    });
+    userEvent.click(button);
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
   };
 
   it('renders a button', () => {
-    const root = render();
+    render();
 
-    expect(root).toHaveClassName('ConfirmButton');
-    expect(root.find(Button)).toHaveLength(1);
-    expect(root.find(Button)).toHaveClassName('ConfirmButton-default-button');
-    expect(root.find(Button)).toHaveProp('buttonType', 'neutral');
+    const button = screen.getByRole('button', {
+      name: 'the default text of this button',
+    });
+    expect(button).toHaveClass('ConfirmButton-default-button');
+    expect(button).toHaveClass('Button--neutral');
   });
 
   it('passes the buttonType prop to the button', () => {
-    const buttonType = 'alert';
-    const root = render({ buttonType });
+    render({ buttonType: 'alert' });
 
-    expect(root.find(Button)).toHaveProp('buttonType', buttonType);
+    const button = screen.getByRole('button', {
+      name: 'the default text of this button',
+    });
+    expect(button).toHaveClass('ConfirmButton-default-button');
+    expect(button).toHaveClass('Button--alert');
   });
 
   it('passes the children prop to the button', () => {
-    const children = 'Do you really want to delete this?';
-    const root = render({ children });
+    render({ children: 'Do you really want to delete this?' });
 
-    expect(root.find(Button).children()).toHaveText(children);
+    const button = screen.getByRole('button', {
+      name: 'Do you really want to delete this?',
+    });
+    expect(button).toHaveClass('ConfirmButton-default-button');
+    expect(button).toHaveClass('Button--neutral');
   });
 
   it('shows ConfirmationDialog when button is clicked', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ store });
+    render();
+    const button = screen.getByRole('button', {
+      name: 'the default text of this button',
+    });
 
-    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(0);
+    expect(
```

Comment:
Nit: Because the code we are testing is adding the class name `ConfirmButton--show-confirmation` to the element with a class name of `ConfirmButton` conditionally, it feels a bit better to me to write this as `expect(screen.getByClassName('ConfirmButton')).not.toHaveClass('ConfirmButton--show-confirmation');`, but ultimately this code accomplishes the same thing.